cpp/arcticdb/version/test/test_version_map.cpp

```diff
@@ -750,48 +767,58 @@ TEST(VersionMap, CacheInvalidation) {
             LoadStrategy{LoadType::FROM_TIME, LoadObjective::INCLUDE_DELETED, static_cast<timestamp>(1)},
-            LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY,
-                         static_cast<timestamp>(2)}, // when include_deleted=false FROM_TIME searches for an undeleted version
-            LoadStrategy{LoadType::LATEST, LoadObjective::UNDELETED_ONLY},
         },
 
         // Different parameters which should all load to v2
         std::vector<LoadStrategy>{
             LoadStrategy{LoadType::DOWNTO, LoadObjective::INCLUDE_DELETED, static_cast<SignedVersionId>(2)},
-            LoadStrategy{LoadType::DOWNTO, LoadObjective::INCLUDE_DELETED, static_cast<SignedVersionId>(-1)},
+            LoadStrategy{LoadType::DOWNTO, LoadObjective::INCLUDE_DELETED, static_cast<SignedVersionId>(-2)},
             LoadStrategy{LoadType::FROM_TIME, LoadObjective::INCLUDE_DELETED, static_cast<timestamp>(2)},
+            LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY,
+                         static_cast<timestamp>(3)}, // when include_deleted=false FROM_TIME searches for an undeleted version
+            LoadStrategy{LoadType::LATEST, LoadObjective::UNDELETED_ONLY},
+        },
+        // Different parameters which should all load to v3
+        std::vector<LoadStrategy>{
+            LoadStrategy{LoadType::DOWNTO, LoadObjective::INCLUDE_DELETED, static_cast<SignedVersionId>(3)},
+            LoadStrategy{LoadType::DOWNTO, LoadObjective::INCLUDE_DELETED, static_cast<SignedVersionId>(-1)},
+            LoadStrategy{LoadType::FROM_TIME, LoadObjective::INCLUDE_DELETED, static_cast<timestamp>(3)},
             LoadStrategy{LoadType::LATEST, LoadObjective::INCLUDE_DELETED},
         }
     };
 
     for (auto i=0u; i<num_versions; ++i){
         for (auto j=0u; j<num_versions; ++j){
-            // For every two versions we check that all load params for earlier versions cache load paramse for later versions:
+            // For every two versions we check that all load params for earlier versions cache load params for later versions:
             check_all_caching(should_load_to_v[i], should_load_to_v[j], i<=j);
         }
 
-        // ALL and LOAD_UNDELETED because they both load to v0 (UNDELETED_ONLY will load v0 because only there it will load
-        check_all_caching({load_all_param, load_all_undeleted_param}, should_load_to_v[i], true);
-        check_all_caching(should_load_to_v[i], {load_all_param, load_all_undeleted_param}, false);
+        // LOAD_ALL loads to version 0
+        check_all_caching({load_all_param}, should_load_to_v[i], true);
+        // Load_all_undeleted loads to version 1 because of the tombstone_all in ver 1
+        check_all_caching({load_all_undeleted_param}, should_load_to_v[i], i != 0);
+
+        // If we have loaded to version 0 all load requests are true
+        check_all_caching(should_load_to_v[i], {load_all_param, load_all_undeleted_param}, i == 0);
```

Comment:
I think we can make `check_all_caching(should_load_to_v[1], {load_all_undeleted_param}, true)` pass?

It will need the comment about `LOAD_ALL` `UNDELETED_ONLY` caching changed to pass.

Reply:
Actually even better, with your changes we can remove the separate testing for `load_all` and `load_all_undeleted`.

Whu don't just place the `LOAD_ALL` in `should_load_to_v[0]`
and `LOAD_ALL_UNDELETED` in `should_load_to_v[1]`?